src/heretic/model.py

```diff
@@ -111,48 +155,47 @@ def reload_model(self):
             self.trusted_models[self.settings.model] = True
 
     def get_layers(self) -> ModuleList:
+        model = self.model
+        
+        # Unwrap PeftModel
+        if isinstance(model, PeftModel):
+            model = model.base_model.model
+
         # Most multimodal models.
         with suppress(Exception):
-            return self.model.model.language_model.layers
+            return model.model.language_model.layers
 
         # Text-only models.
-        return self.model.model.layers
+        return model.model.layers
 
-    def get_layer_matrices(self, layer_index: int) -> dict[str, list[Tensor]]:
+    def get_layer_modules(self, layer_index: int) -> dict[str, list[torch.nn.Module]]:
         layer = self.get_layers()[layer_index]
 
-        matrices = {}
+        modules = {}
 
-        def try_add(component: str, matrix: Any):
-            # Handle Triton tensors (e.g., from MXFP4 quantization) by extracting
-            # the underlying PyTorch tensor via the .data attribute.
-            if hasattr(matrix, "data") and torch.is_tensor(matrix.data):
-                matrix = matrix.data
+        def try_add(component: str, module: Any):
+            if component not in modules:
+                modules[component] = []
 
-            assert torch.is_tensor(matrix)
```

Comment:
We should still have an equivalent check. As it stands, the code will blindly add fields as long as they exist, without verifying that they are tensors or modules.